app/Http/Controllers/MasjidProfileController.php

```diff
@@ -39,4 +40,32 @@ public function update(Request $request): RedirectResponse
 
         return redirect()->route('masjid_profile.show');
     }
+
+    public function image(Request $request)
+    {
+        $this->authorize('edit_masjid_profile');
+
+        $validatedPayload = $request->validate([
+            'image' => 'required'
+        ]);
+
+        if (!base64_decode($validatedPayload['image'])) {
+            return response()->json([
+                'message' => 'Image not found'
+            ]);
+        }
+
+        $image_parts      = explode(";base64,", $request->image);
+        $image_base64     = base64_decode($image_parts[1]);
+        $image_name       = uniqid() . '.webp';
+        $image_full_path  = 'public/' . $image_name;
```

Comment:
Ini `public/` nya hardcodded, agak sulit kita mau pindah Storage driver.
```suggestion
        $image_full_path  = $image_name;
```
Sebaiknya kita manfaatkan storage driver `public` saja. Jadi di `.env` dan `.env.example` kita set seperti ini:
```
FILESYSTEM_DRIVER=public
```